Mage.Sets/src/mage/cards/m/MazesMantle.java

```diff
@@ -0,0 +1,61 @@
+package mage.cards.m;
+
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.condition.common.AttachedToMatchesFilterCondition;
+import mage.abilities.decorator.ConditionalContinuousEffect;
+import mage.abilities.effects.common.AttachEffect;
+import mage.abilities.effects.common.continuous.BoostEnchantedEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.keyword.EnchantAbility;
+import mage.abilities.keyword.FlashAbility;
+import mage.abilities.keyword.HexproofAbility;
+import mage.abilities.keyword.ToxicAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterPermanent;
+import mage.filter.predicate.mageobject.AbilityPredicate;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetCreaturePermanent;
+
+import java.util.UUID;
+
+public class MazesMantle extends CardImpl {
+    public MazesMantle(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{G}");
+        this.addSubType(SubType.AURA);
+
+        //Flash
+        this.addAbility(FlashAbility.getInstance());
+
+        //Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        Ability ability = new EnchantAbility(auraTarget);
+        this.addAbility(ability);
+
+        //When Maze’s Mantle enters the battlefield, if enchanted creature has toxic, that creature gains hexproof until end of turn.
+        FilterPermanent filter = new FilterPermanent("if enchanted creature has toxic");
+        filter.add(new AbilityPredicate(ToxicAbility.class));
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new ConditionalContinuousEffect(
```

Comment:
This needs to be an `ConditionalInterveningIfTriggeredAbility` (e.g. [Howl of the Hunt](https://github.com/magefree/mage/blob/c9a7aedfb2a61a7e24c9bdb1ab723617a6ef2e93/Mage.Sets/src/mage/cards/h/HowlOfTheHunt.java)) or a custom ability with an overridden `checkInterveningIfClause` function (e.g. [Narcolepsy](https://github.com/magefree/mage/blob/c9a7aedfb2a61a7e24c9bdb1ab723617a6ef2e93/Mage.Sets/src/mage/cards/n/Narcolepsy.java))

Reply:
i have implemented this now. But can you explain why it has to be a intervening triggered abillity? in my tests the card worked just fine.

Reply:
Your original implementation had the triggered ability trigger no matter what, then only have the creature gain hexproof if it has toxic. With `ConditionalInterveningIfTriggeredAbility`, the triggered ability only triggers *at all* if the creature has toxic, so attaching it to a vanilla creature causes it to gain +2/+2 and no triggered ability go on the stack